File: JobApplicationMgtSystem/JobApplicationApi/JobApplicationApi/Models/JobApplication.cs
namespace JobApplicationApi.Models
{
    public class JobApplication
    {
        public int Id { get; set; }
        public DateOnly AppliedAt { get; set; } = DateOnly.FromDateTime(DateTime.UtcNow);
        public ApplicationStatus Status { get; set; } = ApplicationStatus.Submitted;
        public string? CoverLetter { get; set; }
        public string? ResumeSnapshotUrl { get; set; }  // resume as it was at time of applying

        // FK to candidate
        public int CandidateProfileId { get; set; }
        public Candidate Candidate { get; set; } = null!;

        // FK to the job they applied for
        public int JobPostingId { get; set; }
        public JobPosting JobPosting { get; set; } = null!;
        public ICollection<InterviewStage> InterviewStages { get; set; } = new List<InterviewStage>();
    }
}

File: JobApplicationMgtSystem/JobApplicationApi/Frontend/jobapp-frontend/src/types/jobApplication.ts
import type { CandidateSummary } from "./candidate";
import type { JobPostingSummary } from "./jobPosting";
import type { InterviewStage } from "./interviewStage";

export interface JobApplication
{
    id: number;
    status: ApplicationStatus;
    appliedAt: string; // DateOnly → "YYYY-MM-DD"
    coverLetter: string;
    resumeSnapshotUrl: string;
    candidate: CandidateSummary;
    jobPosting: JobPostingSummary;
    interviewStages: InterviewStage[];
}

export const ApplicationStatus =
{
    Submitted: 0,
    Interview: 1,
    Offer: 2,
    Rejected: 3,
    Withdrawn: 4,
} as const;

export type ApplicationStatus = typeof ApplicationStatus[keyof typeof ApplicationStatus];

export const applicationStatusLabels: Record<ApplicationStatus, string> = {
  [ApplicationStatus.Submitted]: "Submitted",
  [ApplicationStatus.Interview]: "Interview",
  [ApplicationStatus.Offer]: "Offer",
  [ApplicationStatus.Rejected]: "Rejected",
  [ApplicationStatus.Withdrawn]: "Withdrawn",
};